src/Go/effects/netscriptGoImplementation.ts

```diff
@@ -104,42 +120,53 @@ export function validateTurn(error: (s: string) => void, moveString = "") {
 /**
  * Pass player's turn and await the opponent's response (or logs the end of the game if both players pass)
  */
-export async function handlePassTurn(logger: (s: string) => void) {
-  passTurn(Go.currentGame, GoColor.black);
+export async function handlePassTurn(logger: (s: string) => void, passAsWhite = false) {
+  const color = passAsWhite ? GoColor.white : GoColor.black;
+  passTurn(Go.currentGame, color);
   logger("Go turn passed.");
+  GoEvents.emit();
 
   if (Go.currentGame.previousPlayer === null) {
     logEndGame(logger);
-    return getOpponentNextMove(false, logger);
+    return getOpponentNextMove(false, logger, passAsWhite);
   } else {
-    return makeAIMove(Go.currentGame);
+    return makeAIMove(Go.currentGame, true, passAsWhite);
   }
 }
 
 /**
  * Validates and applies the player's router placement
  */
-export async function makePlayerMove(logger: (s: string) => void, error: (s: string) => void, x: number, y: number) {
+export async function makePlayerMove(
+  logger: (s: string) => void,
+  error: (s: string) => void,
+  x: number,
+  y: number,
+  playAsWhite = false,
+) {
   const boardState = Go.currentGame;
-  const validity = evaluateIfMoveIsValid(boardState, x, y, GoColor.black);
-  const moveWasMade = makeMove(boardState, x, y, GoColor.black);
+  const color = playAsWhite ? GoColor.white : GoColor.black;
+  const validity = evaluateIfMoveIsValid(boardState, x, y, color);
+  const moveWasMade = makeMove(boardState, x, y, color);
 
   if (validity !== GoValidity.valid || !moveWasMade) {
     error(`Invalid move: ${x} ${y}. ${validity}.`);
   }
 
   GoEvents.emit();
-  logger(`Go move played: ${x}, ${y}`);
-  return makeAIMove(boardState);
+  logger(`Go move played: ${x}, ${y}${playAsWhite ? " (White)" : ""}`);
+  return makeAIMove(boardState, true, playAsWhite);
 }
 
 /**
   Returns the promise that provides the opponent's move, once it finishes thinking.
  */
-export async function getOpponentNextMove(logOpponentMove = true, logger: (s: string) => void) {
+export async function getOpponentNextMove(logOpponentMove = true, logger: (s: string) => void, playAsWhite = false) {
+  const nextMovePromise = playAsWhite ? "nextTurnForWhite" : "nextTurn";
```

Comment:
This really solidifies my feeling that these should not be separate variables. Rather, there should be a single `nextTurn` object that is indexed by GoColor, or possibly a two-length array. The same probably makes sense for most of the things you've split out as separate variables.

Reply:
I reworked this to be a GoColor object, which is definitely cleaner